overlord/devicestate/devicemgr.go

```diff
@@ -627,12 +644,19 @@ func (m *DeviceManager) seedStart() (*timings.Timings, error) {
 
 func (m *DeviceManager) preloadGadget() (sysconfig.Device, *gadget.Info, error) {
 	var sysLabel string
-	modeEnv, err := maybeReadModeenv()
-	if err != nil {
-		return nil, nil, err
-	}
-	if modeEnv != nil {
-		sysLabel = modeEnv.RecoverySystem
+	if m.preseed {
+		if m.preseedSystem == "" {
+			return nil, nil, fmt.Errorf("no system to preseed")
```

Comment:
nitpick: we don't expect this to happen if we get this far, right?

maybe we could have the field be called preesedSystemLabel and preseedSystem be an accessor that panics if it is unset?

Reply:
Indeed, added.